File: VB/SpreadsheetSpellChecker.vb
Imports DevExpress.Xpf.Core
Imports DevExpress.Xpf.SpellChecker
Imports DevExpress.Xpf.Spreadsheet
Imports DevExpress.Xpf.Spreadsheet.Internal
Imports DevExpress.XtraSpellChecker

Namespace SpreadsheetSpellchecking_WPF

    Public Class SpreadsheetSpellChecker
        Inherits DXSpellCheckerBase(Of SpreadsheetControl)

        Private ReadOnly Property Spreadsheet As SpreadsheetControl
            Get
                Return AssociatedObject
            End Get
        End Property

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            SpellingSettings.RegisterTextControl(GetType(XpfCellInplaceEditor))
            SubscribeToEvents()
        End Sub

        Protected Overrides Sub OnDetaching()
            UnsubscribeFromEvents()
            SpellingSettings.UnregisterTextControl(GetType(XpfCellInplaceEditor))
            MyBase.OnDetaching()
        End Sub

        Private Sub SubscribeToEvents()
            Me.Spreadsheet.CellEditorOpened += AddressOf OnSpreadsheet_CellEditorOpened
            SpellChecker.CheckCompleteFormShowing += AddressOf OnChecker_CheckCompleteFormShowing
        End Sub

        Private Sub UnsubscribeFromEvents()
            Me.Spreadsheet.CellEditorOpened -= AddressOf OnSpreadsheet_CellEditorOpened
            SpellChecker.CheckCompleteFormShowing -= AddressOf OnChecker_CheckCompleteFormShowing
        End Sub

        Private Sub OnSpreadsheet_CellEditorOpened(ByVal sender As Object, ByVal e As CellEditorOpenedEventArgs)
            If Not e.IsCustom Then
                e.Editor.SetValue(ThemeManager.ThemeNameProperty, ApplicationThemeHelper.ApplicationThemeName)
                SpellChecker.Check(e.Editor)
            End If
        End Sub

        Private Sub OnChecker_CheckCompleteFormShowing(ByVal sender As Object, ByVal e As FormShowingEventArgs)
            e.Handled = True
        End Sub
    End Class
End Namespace

File: CS/SpreadsheetSpellChecker.cs
using DevExpress.Xpf.Core;
using DevExpress.Xpf.SpellChecker;
using DevExpress.Xpf.Spreadsheet;
using DevExpress.Xpf.Spreadsheet.Internal;
using DevExpress.XtraSpellChecker;

namespace SpreadsheetSpellchecking_WPF {
    public class SpreadsheetSpellChecker : DXSpellCheckerBase<SpreadsheetControl> {
        SpreadsheetControl Spreadsheet { get { return AssociatedObject; } }
        protected override void OnAttached() {
            base.OnAttached();
            SpellingSettings.RegisterTextControl(typeof(XpfCellInplaceEditor));
            SubscribeToEvents();
        }
        protected override void OnDetaching() {
            UnsubscribeFromEvents();
            SpellingSettings.UnregisterTextControl(typeof(XpfCellInplaceEditor));
            base.OnDetaching();
        }

        void SubscribeToEvents() {
            Spreadsheet.CellEditorOpened += OnSpreadsheet_CellEditorOpened;
            SpellChecker.CheckCompleteFormShowing += OnChecker_CheckCompleteFormShowing;   
        }
        void UnsubscribeFromEvents() {
            Spreadsheet.CellEditorOpened -= OnSpreadsheet_CellEditorOpened;
            SpellChecker.CheckCompleteFormShowing -= OnChecker_CheckCompleteFormShowing;
        }

        void OnSpreadsheet_CellEditorOpened(object sender, CellEditorOpenedEventArgs e) {
            if (!e.IsCustom)
            {
                e.Editor.SetValue(ThemeManager.ThemeNameProperty, ApplicationThemeHelper.ApplicationThemeName);
                SpellChecker.Check(e.Editor);
            }
        }

        void OnChecker_CheckCompleteFormShowing(object sender, FormShowingEventArgs e)
        {
           e.Handled = true;
        }
    }
}
